features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/FlagsClient.kt

```diff
@@ -10,114 +10,175 @@ import com.datadog.android.Datadog
+     * Set the context for the client.
+     * @param context The evaluation context containing targeting key and attributes.
+     */
+    fun setContext(context: EvaluationContext)
 
-    private val registeredProviders: MutableMap<SdkCore, FlagsProvider> = mutableMapOf()
+    /**
+     * Resolve a Boolean flag value.
+     * @param flagKey The name of the key to query.
+     * @param defaultValue The value to return if the key cannot be retrieved.
+     * @return The value of the flag, or the default value if it cannot be retrieved.
+     */
+    fun resolveBooleanValue(flagKey: String, defaultValue: Boolean): Boolean
 
     /**
-     * Identify whether a [FlagsProvider] has previously been registered for the given SDK instance.
-     *
-     * This check is useful in scenarios where more than one component may be responsible
-     * for registering a provider.
-     *
-     * @param sdkCore the [SdkCore] instance to check against. If not provided, default instance
-     * will be checked.
-     * @return whether a provider has been registered
+     * Resolve a String flag value.
+     * @param flagKey The name of the key to query.
+     * @param defaultValue The value to return if the key cannot be retrieved.
+     * @return The value of the flag, or the default value if it cannot be retrieved.
      */
-    @JvmOverloads
-    @JvmStatic
-    fun isRegistered(sdkCore: SdkCore = Datadog.getInstance()): Boolean = synchronized(registeredProviders) {
-        registeredProviders.containsKey(sdkCore)
-    }
+    fun resolveStringValue(flagKey: String, defaultValue: String): String
 
     /**
-     * Returns the constant [FlagsProvider] instance.
-     *
-     * Until a Flags feature is enabled, a no-op implementation is returned.
-     *
-     * @return The provider associated with the instance given instance, or a no-op provider. If SDK
-     * instance is not provided, default instance will be used.
+     * Resolve a Number flag value.
+     * @param flagKey The name of the key to query.
+     * @param defaultValue The value to return if the key cannot be retrieved.
+     * @return The value of the flag, or the default value if it cannot be retrieved.
      */
-    @JvmOverloads
-    @JvmStatic
-    fun get(sdkCore: SdkCore = Datadog.getInstance()): FlagsProvider = synchronized(registeredProviders) {
-        val provider = registeredProviders[sdkCore]
-        if (provider == null) {
-            val errorMsg = "No FlagsProvider for the SDK instance with name ${sdkCore.name} " +
-                "found, returning no-op implementation."
-            (sdkCore as? FeatureSdkCore)
-                ?.internalLogger
-                ?.log(
-                    InternalLogger.Level.WARN,
-                    InternalLogger.Target.USER,
-                    { errorMsg }
-                )
-            NoOpFlagsProvider()
-        } else {
-            provider
-        }
-    }
+    fun resolveNumberValue(flagKey: String, defaultValue: Number): Number
```

Comment:
question: is it on purpose that we have just a single `Number` method for numbers instead of `resolveIntValue`, `resolveLongValue`, `resolveDoubleValue`, `resolveFloatValue`? so that with current approach the client is supposed to make a down-cast?

upd: ok, there is `resolveIntValue` below though, but nothing for long, float and double.